substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/JfrTypes.java

```diff
@@ -24,22 +24,42 @@
  */
 package com.oracle.svm.jfr;
 
+import jdk.jfr.internal.Type;
+import jdk.jfr.internal.TypeLibrary;
+import org.graalvm.nativeimage.Platform;
+import org.graalvm.nativeimage.Platforms;
+
+import java.util.List;
+
 public enum JfrTypes {
-    // TODO: see JfrEvents, same TODOs apply here as well.
-    Class(20),
-    String(21),
-    StackTrace(23),
-    Method(38),
-    Symbol(39),
-    Frametype(58);
+    Class(getTypeId("java.lang.Class")),
+    String(getTypeId("java.lang.String")),
+    StackTrace(getTypeId("jdk.types.StackTrace")),
+    ClassLoader(getTypeId("jdk.types.ClassLoader")),
+    Method(getTypeId("jdk.types.Method")),
+    Symbol(getTypeId("jdk.types.Symbol")),
+    Module(getTypeId("jdk.types.Module")),
+    Package(getTypeId("jdk.types.Package")),
+    FrameType(getTypeId("jdk.types.FrameType"));
 
-    private final int id;
+    private final long id;
 
-    JfrTypes(int id) {
+    JfrTypes(long id) {
```

Comment:
This constructor could have a `String` as an argument. Then it would be possible to move the `getTypeId(string)` call into the constructor.